src/core/blocks/base_block.c

```diff
@@ -4,28 +4,35 @@
 
 #include "base_block.h"
 
-char *ser_blockheader(char *dest, BlockHeader *block_header){
+unsigned char *ser_blockheader(unsigned char *dest, BlockHeader *block_header){
   memcpy(dest, &(block_header->timestamp), sizeof(block_header->timestamp));
 
-  char *all_tx = dest + sizeof(block_header->timestamp);
+  unsigned char *all_tx = dest + sizeof(block_header->timestamp);
   memcpy(all_tx, &(block_header->all_tx), sizeof(block_header->all_tx));
 
-  char *prev_header_hash = all_tx + sizeof(block_header->all_tx);
+  unsigned char *prev_header_hash = all_tx + sizeof(block_header->all_tx);
   memcpy(prev_header_hash, &(block_header->prev_header_hash), sizeof(block_header->prev_header_hash));
 
-  char *nonce = prev_header_hash + sizeof(block_header->prev_header_hash);
+  unsigned char *nonce = prev_header_hash + sizeof(block_header->prev_header_hash);
   memcpy(nonce, &(block_header->nonce), sizeof(block_header->nonce));
 
-  char* terminate = nonce+sizeof(block_header->nonce);
+  unsigned char* terminate = nonce+sizeof(block_header->nonce);
   return terminate;
 }
 
-char *ser_blockheader_alloc(BlockHeader *block_header){
-  char *data = malloc(sizeof(BlockHeader));
+unsigned char *ser_blockheader_alloc(BlockHeader *block_header){
+  unsigned char *data = malloc(sizeof(BlockHeader));
   ser_blockheader(data, block_header);
   return data;
 }
 
+void hash_blockheader(BlockHeader *header, unsigned char *buf) {
```

Comment:
Should we make _alloc versions of this also? I'm also worried about sizing/safety, should we pass size just to verify that the buffer we give is the right size and we aren't doing a bad operations?

Reply:
I think we're safe as is? Given we only use one hashing function it seems like we can pretty safely put the burden on the caller. mbed tls doesn't take a size for the buffer to write the hash to either.

I do think the comment below of adding docstring to specify the expected length is a good idea, and will help mitigate issues.